doc/introduction/interfaces/numpy.rst

```diff
@@ -110,14 +110,66 @@ with respect to both QNode parameters ``phi`` and ``theta``:
 
     phi = np.array([0.5, 0.1])
     theta = 0.2
-    dcircuit = qml.grad(circuit3, argnum=[0, 1])
+    dcircuit = qml.grad(circuit3)
 
 Evaluating this gradient function at specific parameter values:
 
 >>> dcircuit(phi, theta)
 (array([ -4.79425539e-01,   1.11022302e-16]), array(0.0))
 
 
+Non-differentiable arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Note that **all positional arguments provided to the QNode are assumed to be differentiable
+by default**:
```

Comment:
```suggestion
by default, while keyword arguments are not.**:
```

Reply:
Is that even true?

Reply:
Keyword arguments can never be differentiable in the current system :) I would like to change that, so that the type of argument no longer affects differentiability at all (hence why I never mentioned keyword args here, I did that somewhat deliberately)